ports/esp32/modsmartconfig.c

```diff
@@ -0,0 +1,248 @@
+#include <string.h>
+
+#include "py/runtime.h"
+
+#include "freertos/FreeRTOS.h"
+#include "freertos/task.h"
+#include "freertos/event_groups.h"
+
+#include "esp_wifi.h"
+#include "esp_event.h"
+#include "esp_log.h"
+#include "esp_smartconfig.h"
+
+#include "mphalport.h"
+#include "modsmartconfig.h"
+
+static EventGroupHandle_t wifi_event_group;
+
+static bool smartconfig_process_done = false;
+
+// smartconfig info variables
+static uint8_t ssid[33] = {0};
+static uint8_t bssid[6] = {0};
+static uint8_t password[65] = {0};
+static uint8_t rvd_data[33] = {0};
+static int8_t type = SC_TYPE_ESPTOUCH_AIRKISS;
```

Comment:
Is there a reason why a couple of these arrays (`ssid`, `password`) [have an extra byte](https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-reference/network/esp_smartconfig.html#_CPPv433smartconfig_event_got_ssid_pswd_t)?

Reply:
I made the following attempt:

1. Create a new cell phone hotspot

2. Setup `ssid` as long as possible (my phone can set 32 characters, but my colleague's phone can only set 30 characters)

3. Go through the `smartconfig` process, and connected to the hotspot correctly

4. Use `smartconfig.ssid()` (or other functions returned with variable length) to return the obtained ssid

    1. If the length of the hotspot ssid is **less than 32**, then the correct contents will be printed
    2. If the length of the hotspot ssid is **equals to 32**, then it will print more than 32 characters, as for the exact length, I guess it needs to find the next `\0` character?

Reply:
These attempts presuppose that the array size is set to the same size as the array declared in the document, without an extra byte.